DEVELOPING_OPAL/tools/src/main/scala/org/opalj/support/info/Purity.scala

```diff
@@ -479,235 +577,60 @@ object Purity {
-            case None | Some("L1") =>
-                support ::= LazyInterProceduralEscapeAnalysis
-
-            case Some("none") =>
-
-            case Some(a) =>
-                Console.println(s"unknown escape analysis: $a")
-                Console.println(usage)
-                return;
-        }
-
-        fieldAssignabilityAnalysisName match {
-            case Some("L0") if eager => support ::= EagerL0FieldAssignabilityAnalysis
-
-            case Some("L0") => support ::= LazyL0FieldAssignabilityAnalysis
-
-            case Some("L1") if eager => support ::= EagerL1FieldAssignabilityAnalysis
-
-            case Some("L1") => support ::= LazyL1FieldAssignabilityAnalysis
-
-            case Some("L2") if eager =>
-                support ::= EagerL2FieldAssignabilityAnalysis
-
-            case Some("L2") =>
-                support ::= LazyL2FieldAssignabilityAnalysis
-
-            case Some("none") =>
-
-            case None => analysis match {
-                    case LazyL0PurityAnalysis => support ::= LazyL0FieldAssignabilityAnalysis
-                    case LazyL1PurityAnalysis => support ::= LazyL1FieldAssignabilityAnalysis
-                    case LazyL2PurityAnalysis => support ::= LazyL1FieldAssignabilityAnalysis
-                }
-
-            case Some(a) =>
-                Console.println(s"unknown field assignability analysis: $a")
-                Console.println(usage)
-                return;
-        }
-
-        val d =
-            if (domainName.isEmpty)
-                classOf[domain.l2.DefaultPerformInvocationsDomainWithCFGAndDefUse[_]]
-            else {
-                Class.forName(domainName.get).asInstanceOf[Class[Domain with RecordDefUse]]
-            }
-
-        val rater = if (raterName.isEmpty) {
-            SystemOutLoggingAllExceptionRater
-        } else {
-            import scala.reflect.runtime.universe.runtimeMirror
-            val mirror = runtimeMirror(getClass.getClassLoader)
-            val module = mirror.staticModule(raterName.get)
-            mirror.reflectModule(module).instance.asInstanceOf[DomainSpecificRater]
-        }
-
-        val callGraphKey = callGraphName match {
-            case Some("CHA")        => CHACallGraphKey
-            case Some("PointsTo")   => AllocationSiteBasedPointsToCallGraphKey
-            case Some("RTA") | None => RTACallGraphKey
-            case Some(a) =>
-                Console.println(s"unknown call graph analysis: $a")
-                Console.println(usage)
-                return;
-        }
-
-        if (evaluationDir.isDefined && !evaluationDir.get.exists()) evaluationDir.get.mkdir
-
         val begin = Calendar.getInstance()
         Console.println(begin.getTime)
 
         time {
-            if (multiProjects) {
-                for (subp <- cp.listFiles().filter(_.isDirectory)) {
+            if (purityConf.multiProjects) {
+                for (subp <- purityConf.classPathFiles.head.listFiles().filter(_.isDirectory)) {
```

Comment:
Using `.head` here (always) seems strange. What would be the semantics if there were multiple files given?